pkg/nsx/services/inventory/inventory.go

```diff
@@ -0,0 +1,243 @@
+	// If there is no such cluster, create one.
+	// Otherwise, sync with NSX for different types of inventory objects.
+	if err == nil {
+		err = s.clusterStore.Add(&cluster)
+		if err != nil {
+			log.Error(err, "Add cluster to store")
+		}
+		return err
+	}
+	log.Error(err, "Cannot find existing container cluster, will create one")
+	newContainerCluster := s.BuildInventoryCluster()
+	cluster, err = s.AddContainerCluster(newContainerCluster)
+	if err != nil {
+		return err
+	}
+	err = s.clusterStore.Add(&cluster)
+	if err != nil {
+		log.Error(err, "Add cluster to store")
+		return err
+	}
+	log.Info("A new ContainerCluster is added", "cluster", newContainerCluster.DisplayName)
+	err = s.initContainerApplicationInstance(s.NSXConfig.Cluster)
+	if err != nil {
+		log.Error(err, "Populating inventory object from nsx error")
+	}
+
+	return nil
+}
+
+func (s *InventoryService) SyncInventoryStoreByType(clusterId string) error {
+	log.Info("Populating inventory object from NSX")
+	err := s.initContainerApplicationInstance(clusterId)
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *InventoryService) SyncInventoryObject(bufferedKeys sets.Set[InventoryKey]) (sets.Set[InventoryKey], error) {
+	retryKeys := sets.New[InventoryKey]()
+	startTime := time.Now()
+	defer func() {
+		log.Info("Finished syncing inventory object", "duration", time.Since(startTime))
+	}()
+	for key := range bufferedKeys {
+		log.V(1).Info("Syncing inventory object", "object key", key)
+		namespace, name, err := cache.SplitMetaNamespaceKey(key.Key)
+		if err != nil {
+			log.Error(err, "Failed to split meta namespace key", "key", key)
+			continue
+		}
+		switch key.InventoryType {
+
+		case ContainerApplicationInstance:
+			retryKey := s.SyncContainerApplicationInstance(name, namespace, key)
+			if retryKey != nil {
+				retryKeys.Insert(*retryKey)
+			}
+		}
+	}
+
+	err := s.sendNSXRequestAndUpdateInventoryStore()
+	if err != nil {
+		return bufferedKeys, err
+	}
+
+	return retryKeys, err
+}
+
+func (s *InventoryService) DeleteResource(externalId string, resourceType InventoryType) error {
+	log.V(1).Info("Delete inventory resource", "resource_type", resourceType, "external_id", externalId)
+	var inventoryObject interface{}
+	exists := false
+	switch resourceType {
+
+	case ContainerApplicationInstance:
+		inventoryObject = s.applicationInstanceStore.GetByKey(externalId)
+		if inventoryObject != nil {
+			exists = true
+		}
+	default:
+		return fmt.Errorf("unknown resource_type : %v for external_id %s", resourceType, externalId)
+	}
+
+	if exists {
+		deletedInfo := make(map[string]interface{})
+		deletedInfo["resource_type"] = resourceType
+		deletedInfo["external_id"] = externalId
+		s.requestBuffer = append(s.requestBuffer, containerinventory.ContainerInventoryObject{ContainerObject: deletedInfo, ObjectUpdateType: operationDelete})
+		s.pendingDelete[externalId] = inventoryObject
+
+		if resourceType == ContainerApplicationInstance {
```

Comment:
Condition is always true

Reply:
no, it needs to support other type inventory